mesonbuild/dependencies/base.py

```diff
@@ -2307,12 +2287,9 @@ def find_external_dependency(name, env, kwargs):
     # display the dependency name with correct casing
     display_name = display_name_map.get(lname, lname)
 
-    # if this isn't a cross-build, it's uninteresting if native: is used or not
-    if not env.is_cross_build():
-        type_text = 'Dependency'
-    else:
-        type_text = 'Native' if kwargs.get('native', False) else 'Cross'
-        type_text += ' dependency'
+    for_machine = MachineChoice.BUILD if kwargs.get('native', False) else MachineChoice.HOST
+
+    type_text = PerMachine('Build-time', 'Run-time', 'Emit-time')[for_machine] + ' dependency'
```

Comment:
Again, I'm not sure this is a super hot path, but it seems like a runtime cost to save 5 lines of typing.

Reply:
Same reason as above, `MachineChoice` *could* actually change, so I prefer the exhaustiveness check.